proxy/grpcproxy/watch.go

```diff
@@ -234,8 +237,35 @@ func (wps *watchProxyStream) sendLoop() {
 			if !ok {
 				return
 			}
-			if err := wps.stream.Send(wresp); err != nil {
-				return
+			// The grpc package's defaultServerMaxSendMessageSize, defaultServerMaxReceiveMessageSize
+			// and the maxSendMesgSize are set to lower numbers in order to demonstrate
+			// the fragmenting
+			maxEventsPerMsg := 20
+			watchRespSent := false
+			for !watchRespSent {
+				if maxEventsPerMsg == 0 {
+					return
+				}
+				for _, fragment := range v3rpc.FragmentWatchResponse(maxEventsPerMsg, wresp) {
```

Comment:
Since the proxy receives responses from [a watch client](https://github.com/coreos/etcd/blob/master/proxy/grpcproxy/watch_broadcast.go#L62), by the time the proxy gets the watch responses that it is supposed to relay to its watchers, the watch response fragments have already been glued back together. If I am not mistaken the data flow seems to be:
mvcc -> server -> client watcher started by proxy -> proxy -> user's client watcher.